components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowConsumer.java

```diff
@@ -196,11 +200,51 @@ private void sendResponse(HttpServerExchange httpExchange, Exchange camelExchang
                 IOHelper.copy(input, output, IOHelper.DEFAULT_BUFFER_SIZE, true);
             }
         } else {
-            TypeConverter tc = getEndpoint().getCamelContext().getTypeConverter();
-            ByteBuffer bodyAsByteBuffer = tc.mandatoryConvertTo(ByteBuffer.class, body);
+            ByteBuffer bodyAsByteBuffer = this.convertBodyToByteBuffer(camelExchange, httpExchange, body);
             httpExchange.getResponseSender().send(bodyAsByteBuffer);
         }
     }
+    
+    /*
+     * when an HTTP response has status code 200 
+     * but does not have content in the body change the status code to 204
+     */
+    private void changeStatusCodeToNoContent(Exchange camelExchange, HttpServerExchange httpExchange) {
+        Message answer = camelExchange.getMessage();
+        if (httpExchange.getStatusCode() == 200) {
+            answer.getHeaders().put(Exchange.HTTP_RESPONSE_CODE, 204);
+            answer.getHeaders().put(Exchange.HTTP_RESPONSE_TEXT, "No Content");
+
+            httpExchange.setStatusCode(204);
+        }
+    }
+ 
+    private ByteBuffer convertBodyToByteBuffer(Exchange camelExchange, HttpServerExchange httpExchange, Object bodyObj) throws NoTypeConversionAvailableException {
+        
+        TypeConverter tc = getEndpoint().getCamelContext().getTypeConverter();
+        ByteBuffer bodyAsByteBuffer = tc.mandatoryConvertTo(ByteBuffer.class, bodyObj);
+        
+        if (!bodyAsByteBuffer.hasRemaining()) {
+            this.changeStatusCodeToNoContent(camelExchange, httpExchange);
+        } else {
+            if (this.hasNoContent(StandardCharsets.UTF_8.decode(bodyAsByteBuffer.asReadOnlyBuffer()).toString())) {
+                this.changeStatusCodeToNoContent(camelExchange, httpExchange);
+            }
+        }
+        
+        return bodyAsByteBuffer;
+
+    }
+    
+    private boolean hasNoContent(String bodyAsString) {
```

Comment:
@CodeSmell You don't need to return true or false, you can just directly return the result of the predicate directly, e.g:
```suggestion
    private boolean hasNoContent(String bodyAsString) {
        return (bodyAsString.trim().isEmpty()
            || bodyAsString.equalsIgnoreCase("No Content")
            || bodyAsString.equalsIgnoreCase("No Body"));
    }
```